Panel Options: link to an open category section

lean257 opened the pull request:
Fixes #40542 
![screen-recorder-tue-oct-19-2021-16-04-55](https://user-images.githubusercontent.com/17167829/137983651-917560f7-d84e-46ef-b290-4d9d73919e3c.gif)

ryantxu committed:
Geomap: use short name for marker selection (#40541)

kminehart committed:
Packaging: Dependency updates (#40534)

* update loki

* install openssl from alpine's edge repository

* extracttraceid moved to tracing package

* remove exit if warning limit is exceeded

* disable flaky libraryelements test

nmarrs committed:
Canvas: Improve enter / exit editor UX (#40591)

torkelo committed:
RowsToFields: Fix issue with field names that are numbers (#40580)

* RowsToFields: Fix issue with field names that are numbers

* Only add the index accessor if field name does not conflict with it

* fix lint

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

thisisobate committed:
Docs: update frontend style guide to highlight new function paradigm (#40347)

* docs: update frontend style guide to highlight new function paradigm

* docs: add section for default exports

* chore: some typo/syntax/grammar fixes

* chore: change new additions from rule to recommendation

* small typo fix

torkelo committed:
Table: Tweak to type icon styles (#40596)

Commit:
Live: Test Converter tab (#40536)

axelavargas committed:
Graph: Fix annotation description not being shown on hover (#40581)

Elfo404 committed:
Chore: Fix remaining ts strict error in Elasticsearch datasource (#40605)

aocenas committed:
Tempo: Rename metrics for service graph (#40584)

* Rename tempo -> traces

* Fix tests

Co-authored-by: Ivana <ivana.huckova@gmail.com>

gabor committed:
prometheus: monaco query editor: use new api from grafana-ui (#40466)

joshhunt committed:
Grafana-UI: Move Select container styles to outer most container (#40611)

* Grafana-UI: Move Select container styles to outer most container

* Update packages/grafana-ui/src/components/Select/Container.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Rename file

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

torkelo committed:
ValueMappings: Fix so that value mappings work in stat/gauge/bargauge/piechart (#40612)

gamab committed:
AccessControl: keyword scope resolution (#40229)

ivanahuckova committed:
Interval: Use gtime for parsing of time string to duration (#40624)

torkelo committed:
Scale: Fixes handling of NaN percent when data min = data max (#40622)

Commit:
Azure Monitor: Add new starter dashboards (#39876)

* Adding more starter dashboards for Azure Monitor data source

* modified dashboard format to match grafana 3.1+ for panel datasource variable templating

* Minor fix to dashboard config

* Getting rid of individual links to dropdown and adding dropdown

* Minor config changes

* Readding dashboard linking and

* Renamed dashboards, added descriptios and exported them properly for external sharing

ashharrison90 committed:
Variables: Improve display when selecting variables with the same value (#40607)

* Variables: Allow selection of different variables with the same value

* Variables: Don't modify this key

jewbetcha committed:
Slider: Adjusts input to enforce min/max values on blur (#40524)

* [cr] adjusts slider input to enfore min/max values on blur

* [cr] additional test for blur action, min/max readability

mckn committed:
Toolkit: Prevent toolkit from crashing when an error occur. (#40627)

dsotirakis committed:
Chore: Make test-frontend step depend on initialize step (#40616)

* Make test-frontend step depend on initialize

Debug

* Clean git tree dir

* Make build-frontend depend on lint-frontend

* Add actual steps back

fredr committed:
Fix 404 link to time series doc (#40565)

dsotirakis committed:
Add /packaging/**/*.tar.gz.sha256 to .gitignore (#40651)

pkolyvas committed:
Docs: Updated docker image names for consistency (#40426)

* Updated the docker image names and added a note

* Updated the example version to current

Commit:
scroll to view with panel query

Commit:
change comment

Commit:
Remove flakiness of google chat tests (#40592)

Commit:
not export const

ryantxu commented:
There is something a bit funky about this solution... when we click expand, it triggers a new render loop that will then change the scroll position:
![with-scroll-to](https://user-images.githubusercontent.com/705951/138001484-71ad7306-7408-4d26-af5d-c19bffd69d83.gif)

Without this PR, the behavior feels more reasonable for click:
![Peek 2021-10-19 15-55](https://user-images.githubusercontent.com/705951/138001667-7cb42029-56f5-4b5d-8d2e-dc6465cf7659.gif)
 
are we able to skip the scroll behavior if we did the click?

nikki-kiga committed:
Geomap: Avoid exceptions when missing data and values (heatmap and zoom settings)

nikki-kiga committed:
Geomap: Add GeoJSON gazetteer (#40589)


Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

torkelo committed:
minor changes

Clarity-89 committed:
QueryGroup: Add missing keys for extra group action components (#40619)

andresmgot committed:
Chore: remove strict errors for cloud-monitoring (#40620)

dsotirakis committed:
Remove git reset - replace with git stash (#40669)

btasker committed:
InfluxDB: Fix spelling - "chronograf" rather than "chronograph" (#40587)

jackw committed:
Chore: Replace Sass-lint with Stylelint (#40617)

* chore(sass): introduce stylelint

* chore(sass): introduce stylelint config that closely matches sass-lint config

* chore: remove sass-lint from codebase

* style(theme): disable stylelint string-quotes rule

* chore(sass): introduce stylelint-prettier-config to prevent config conflicts

* chore: delete sass-lint file

* style(sass): fix remaining stylelint errors

andresmgot committed:
CloudMonitoring: Allow to manually set filters (#40606)

jackw committed:
Storybook: fix theme colors not updating (#40679)

ifrost committed:
Logs: Move button to load logs volume to logs options header (#40578)

* Move button to load logs volume to logs options header

* Update snapshot

* Do not show load logs volume button when data source does not support it

Commit:
Access Control: Add service accounts (#38994)

* Add extra fields to OSS types to support enterprise

* Create a service account at the same time as the API key

* Use service account credentials when accessing API with APIkey

* Add GetRole to service, merge RoleDTO and Role structs

This patch merges the identical OSS and Enterprise data structures, which improves the code for two reasons:

1.  Makes switching between OSS and Enterprise easier
2.  Reduces the chance of incompatibilities developing between the same functions in OSS and Enterprise

* If API key is not linked to a service account, continue login as usual

* Fallback to old auth if no service account linked to key

* Add CloneUserToServiceAccount

* Adding LinkAPIKeyToServiceAccount

* Handle api key link error

* Better error messages for OSS accesscontrol

* Set an invalid user id as default

* Re-arrange field names

* ServiceAccountId is integer

* Better error messages

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>
Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>
Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>
Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>
Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

domasx2 committed:
fix unified alerting enabled checks (#40693)

JohnnyQQQQ committed:
docs: fix typos in unified alerting (#40696)

zserge committed:
Chore: Replace goconvey in multildap package (#40681)

zserge committed:
Chore: Goconvey migration for cleanup package (#40547)

axelavargas committed:
Plugin Catalog: Fix A11y issues on plugin list (#40629)

kaydelaney committed:
A11y/Dashboard: Fix various dashboard settings issues (#40641)

See #39429

kminehart committed:
update dockerfile (#40699)

mckn committed:
DashGrid: Fixed auto resize issue with tabel panel (#40688)

* Changing how we use the AutoSizer component in the DashGrid.

* updated snapshot.

* fixed failing test.

andresmgot committed:
Chore: remove strict errors for azure-monitor (#40675)

Commit:
not useQueryParams

Commit:
not scroll on toggle

Commit:
Revert "not scroll on toggle"

This reverts commit 92e13676976e4be49f678561a28b0b5ac5c17537.
